src/util/tab-events.js

```diff
@@ -23,12 +38,15 @@ export function whenPageDOMLoaded({tabId}) {
     //         },
     //     })
     // })
-
-    // Workaround: we run a script at 'document_end' (= 'interactive')
-    return browser.tabs.executeScript(tabId, {
-        code: 'undefined', // Bogus code, anything works.
-        runAt: 'document_end',
-    })
+    return new Promise((resolve, reject) => {
+        // Workaround: we run a script at 'document_end' (= 'interactive')
+        browser.tabs.executeScript(tabId, {
+            code: 'undefined', // Bogus code, anything works.
+            runAt: 'document_end',
+        }).then(resolve());
```

Comment:
Also, there's no catch and this is discourage (and will be disallowed in node soon)